### DISCUSSION TITLE
Add support for country regions

### DISCUSSION THREAD

**Discussion by Nielsvanpach** [spatie/holidays#14] [Ideas]
As stated in https://github.com/spatie/holidays/pull/8 and https://github.com/spatie/holidays/pull/12 , countries can have region bound holidays.

**Comment by RohanSakhale** [comment]
#12 made an attempt towards supporting region, feedback appreciated

**Reply by jigar-dhulla** [reply]
Suggest to use ISO Standard for states.
https://www.iso.org/obp/ui/#iso:code:3166:IN

**Comment by marlonbasten** [comment]
I added state support and added States for Germany and Austria. #37

**Comment by Nielsvanpach** [comment]
We've added documentation on how we would implement regions, states, departments, ...
https://github.com/spatie/holidays?tab=readme-ov-file#contributing-a-new-country

**Reply by marlonbasten** [reply]
As described in the PR #37 this implementation is not suitable for every use-case.

Example: Spain has national holidays, regional holidays, state holidays, etc...

Commonly, holidays for those regions are listed separately in calendars for example. But when trying to know if a certain day is a holiday, every layer of regions get's checked.

Having this in mind, I don't see a nice way to obtain all that information in the currently proposed way.

There are some good alternative ways to get that done:
https://github.com/spatie/holidays/pull/37#issuecomment-1898632750
https://github.com/spatie/holidays/pull/37#issuecomment-1899182451

Those proposals are not perfect either, but with a bit of tinkering we could get good results.

I, and I think a lot of other people would be very open to tinker on this, but my PR hasn't gotten a reply since you asking for our opinion on that.

If no alternative solution is wanted, you can close my PR. I know that this package is not your main focus and I am very grateful for this package attempt and all your other packages. But this one-sided communication is very frustrating. Nobody adopted this implementation yet because there are too many open questions.

**Reply by Nielsvanpach** [reply]
@marlonbasten, I appreciate your engagement and the time you've invested. Please bear in mind, this package has received over 160 PR in less than a week. As an open-source maintainer, my (free) time is limited, and I don’t feel I need to rush here.

I've considered your suggestions and discussed this with my open source experienced colleagues. We believe that a more effective approach for maintaining regions in this package is already described in the README. But I’ve already communicated this on your PR in an early phase.

**Reply by herpaderpaldent** [reply]
Hey @Nielsvanpach i have tried to follow the README and created a small proof of concept implementing regional support: https://github.com/spatie/holidays/pull/173 

However i was not able to use named parameters with `Country::make()` as this resulted in an error. 

Maybe you can give me some feedback on how you like it?
